src/hooks/useOnClickOutside.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect } from 'react';
+
+// from https://usehooks.com/useOnClickOutside/
+export function useOnClickOutside(
+  ref: React.RefObject<HTMLElement>,
+  handler: Function
+) {
+  useEffect(
+    () => {
+      //@ts-ignore
+      const listener = event => {
+        // Do nothing if clicking ref's element or descendent elements
+        if (!ref.current || ref.current.contains(event.target)) {
```

Comment:
@jb-san I was a bit worried about the browser support for `.contains`. But after some digging it seems to be supported for the last couple of years.